src/System.CommandLine/Invocation/InvocationExtensions.cs

```diff
@@ -184,6 +186,24 @@ public static CommandLineBuilder CancelOnProcessTermination(this CommandLineBuil
             return builder;
         }
 
+        public static CommandLineBuilder UseTypoCorrections(
+            this CommandLineBuilder builder, int maxLevenshteinDistance = 3)
```

Comment:
@Keboo Where did the default of 3 come from? I'm not saying it is a bad default. I am just curious.

Reply:
Mostly from [here](https://www.xkcd.com/221/) and the simplest thing to get working. There is probably some better heuristics to use rather than just picking a number out of a hat.
I considered using something like:
```C#
string unmatchedToken = ...;
int maxLevenshteinDistance = (int)Math.Ceiling(unmatchedToken.Length / 2.0);
```
Open to other suggestions too.

The other part I am not completely happy with in single character aliases. You can imaging have a command with options `-a`, `-b`, and `-c` and then the user types `-d`. All of those would show as suggestions since they all have the same distance.

Reply:
Certainly. I think there is an outstanding question of what is expected. I think for single character tokens it might be reasonable to either not provide any suggestions or do something different (perhaps only return suggestions for things that start with the same character as the token). I am not really sure what would be expected or desired.

Reply:
I wonder if there should be more consistency of behavior here with tab suggestions. There, we do substring matching, which is a different kind of fuzzy matching. If someone types a single letter and hits tab, they get back all of the symbols containing that letter. That could work for typo suggestions as well.

Similarly, if someone types `myapp --nfo⇥`, maybe suggestions should allow tabbing through the typo-corrected variants?

In other words, is typo correction just a different interaction model for the existing `ISuggestionSource` abstractions, which should be expanded to consider misspellings in addition to the existing substring matching?